CHANGELOG.md

```diff
@@ -56,6 +56,18 @@ SPDX-License-Identifier: LGPL-3.0-or-later
 - Add an About Ikarus page in the documentation ([#291](https://github.com/ikarus-project/ikarus/pull/291))
 - Add new class `Vtk::Writer`, which implements some convenience methods over the existing `dune-vtk` module ([#309](https://github.com/ikarus-project/ikarus/pull/309))
 - Add `VanishingStrain` material (useful for example for plane strain case), also refactor the constructor of `LinearElastic` to take any linear material law ([#317](https://github.com/ikarus-project/ikarus/pull/317))
+- Refactor EAS to handle NonLinearElastic ([#325](https://github.com/ikarus-project/ikarus/pull/325))
+    - `updateState()` function is added to `mixin.hh`, which can be used to update the internal state variables of the skills.
+        - `updateStateImpl()` now has to be implemented by every skill.
+        - For EAS, `updateStateImpl()` is used to update the internal variable `alpha` in a nonlinear analysis.
+    - Synchronize functions like `internalEnergy`, `stress` and `materialTangent` between `LinearElastic` and `NonLinearElastic`.
+    - A `helperfunctions.hh` file is added that contains the helper functions used by the nonlinear solvers.
+        - This contains the function `updateStates()` that calls the `updateState()` function for every finite element.
+        - `updateStates()` is called by nonlinear solvers during every iteration, whenever the correction vector is updated.
+    - `NonLinearOperator` now has an additional template argument, `Assembler`.
```

Comment:
IN my view the Nonlinear operator is by design only a collection of a function and its derivative.
There should be no template or object of an assembler. What should this be doing if all functions are just lambdas without any underlying assembler? But somehow I also dont see a nice solution to this we should chat about different posibilities of design choices. I think it can be already done with observers but this might be to indirect. E.g. all EAS element observe the non-linear solver and as soon as the solvers signals that the iteration or load step is finished the elements get notified and can update what they want. This seems nicer, since we dont have to refactor the design of nonlinearoperator.

Another thing would be a new class that contains a nonlinearoperator **and** the assembler.
Something similar I have done already in python branch for the solvers but Im not srue if we want that.